src/Kernel/ProcessorScheduler.class.st

```diff
@@ -176,6 +176,23 @@ ProcessorScheduler >> highestPriority: newHighestPriority [
 	quiescentProcessLists := newProcessLists
 ]
 
+{ #category : #private }
+ProcessorScheduler >> interpriorityYield: modifiedProcess [
+
+	"The original yield only works within processes of the same priority.
+	The primitive does not try to resume a process in a higher priority.
+	By forcing the creation of a new process and suspending this one, 
+	we force the rescheduling of processes. 
+	
+	This message should not be used directly. 
+	The only application so far is when changing the priority of a process.
+	"
+
+	modifiedProcess isActiveProcess ifFalse: [ ^ self ].
+	[ modifiedProcess resume ] fork.
```

Comment:
I wonder if there can be a corner case where forked process will be executed before the following #suspend. The rescheduling can be triggered  just after the fork by other processes/events in the image. 
I think #fork needs to be done with higher priority than the modifiedProcess to avoid it.

Reply:
Probably, but I think we should build a test for such a thing.
So far, there is not really much code that changes the priority of the running process dynamically, so the question that comes from there is "is it worth it?"